Tensile/Source/lib/include/Tensile/CachingLibrary.hpp

```diff
@@ -157,34 +157,36 @@ namespace Tensile
     {
     public:
         using Library = SolutionLibrary<MyProblem, MySolution>;
-        using Cache   = CacheMap<std::shared_ptr<MySolution>, AMDGPU, MyProblem>;
+        using Cache   = CacheMap<std::tuple<std::shared_ptr<MySolution>, double>, AMDGPU, MyProblem>;
 
         CachingLibrary(std::shared_ptr<Library> subLibrary)
             : m_subLibrary(subLibrary)
-            , m_cache(nullptr)
+            , m_cache(std::make_tuple(nullptr, std::numeric_limits<double>::max()))
         {
         }
 
-        virtual std::shared_ptr<MySolution>
-            findBestSolution(MyProblem const& problem, Hardware const& hardware) const override
+        virtual std::tuple<std::shared_ptr<MySolution>, double>
+            findBestSolutionWithFitness(MyProblem const& problem, Hardware const& hardware) const override
         {
             try
             {
                 auto const& amdgpu = dynamic_cast<AMDGPU const&>(hardware);
 
-                auto rv = m_cache.find(problem, amdgpu);
+                double fitness = std::numeric_limits<double>::max();
+                std::shared_ptr<MySolution> rv;
```

Comment:
`rv` stands for return value.  Maybe this variable should be renamed to 'solution'?